tests/v2/validation/provisioning/rke2/etcd_backup_restore.go

```diff
@@ -0,0 +1,136 @@
+package rke2
+
+import (
+	"fmt"
+
+	"github.com/rancher/norman/types"
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/machinepools"
+)
+
+const (
+	localClusterName             = "local"
+	w1Name                       = "w1"
+	ingress1Name                 = "ingress1"
+	w1ServiceName                = "w1-svc"
+	w2Name                       = "w2"
+	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+)
+
+func createSnapshot(client *rancher.Client, clustername string, generation int, namespace string) error {
+	clusterObj, existingSteveAPIObj, err := getProvisioningClusterByName(client, clustername, namespace)
+	if err != nil {
+		return err
+	}
+
+	clusterObj.Spec.RKEConfig.ETCDSnapshotCreate = &rkev1.ETCDSnapshotCreate{
+		Generation: generation,
+	}
+
+	_, err = client.Steve.SteveType(clusters.ProvisioningSteveResouceType).Update(existingSteveAPIObj, clusterObj)
```

Comment:
Out of curiosity, why are we updating the SteveType object here?

Reply:
I believe since the snapshot is a modification to the cluster, the SteveType object needs to be updated accordingly. Similar to how if we modify a cluster's config, we would need to update that Norman or SteveType object as well.